frontend/integration-tests/tests/crud.scenario.ts

```diff
@@ -334,6 +334,7 @@ describe('Kubernetes resource CRUD operations', () => {
     beforeAll(async () => {
       await browser.get(`${appHost}/k8s/ns/${testName}/${plural}/~new`);
       await yamlView.isLoaded();
+      await browser.wait(until.visibilityOf(yamlView.sideBar));
```

Comment:
Should this be part of `yamlView.isLoaded()`?

Note that the sidebar is hidden at certain screen widths. (I don't know if that will actually be a problem.)

Reply:
If I simply make that line part of `yamlView.isLoaded()`, some CRUD tests start to fail with the browser width specified for protractor runs, because the sidebar on those pages stays closed on load. I will test with `until.presenceOf` in `yamlView.isLoaded` and see if that suffices.

Reply:
> If I simply make that line part of `yamlView.isLoaded()`, some CRUD tests start to fail with the browser width specified for protractor runs, because the sidebar on those pages stays closed on load.

I guess I'm confused then since we're already checking it here. Wouldn't it fail here?

Is there a better element to check for? We're not actually interacting with the sidebar in these tests. I think it might break some scenarios where resources don't have a schema and the sidebar isn't used.

Reply:
Honestly, I've been making sure only "e2e" suite pass. So, I don't know if this has a negative impact in other scenarios. If I increase the `browser.sleep()` call already at the end of `yamlView.isLoaded()` from 1000 to 10000, that will do, but I suspect that's not what you are looking for. Any suggestions as to what element on the page to wait for? If not, I think the current approach to call additional `until.visibilityOf(sidebar)` is flexible and isn't too bad...

Reply:
Does it work if we check for the presence of `.monaco-editor`? We might be able to get rid of the `browser.sleep()` altogether.

Again I have two concerns:

1. We should consolidate whatever logic we decide on in `yamlView.isLoaded()` instead of duplicating it everywhere, particularly since it's not needed for Chrome. Otherwise new tests without this check will randomly fail in Firefox.
2. Some pages don't have a sidebar (like the Import YAML page).

Reply:
I will test it with `presenceOf('.monaco-editor')`, and if works, will also test without the `sleep` in both browsers, and come back here.